File: rust/src/insertion_sort.rs
// Code originally forked from https://github.com/diptangsu/Sorting-Algorithms/
include!("constants.rs");
use std::time::Instant;

pub fn insertion_sort<T: Ord>(array: &mut [T]) {
    for i in 1..array.len() {
        let mut j = i;
        while j > 0 && array[j] < array[j - 1] {
            array.swap(j, j - 1);
            j = j - 1;
        }
    }
}

fn main() {
    unsafe {
        for _x in 0..50 {
            let start = Instant::now();
            insertion_sort(&mut SET1);
            let elapsed = start.elapsed();
            println!("{:?}", elapsed);
        }
    }
}

File: python/insertion_sort.py
# Code originally forked from https://github.com/diptangsu/Sorting-Algorithms/

import time
from constants import *


def insertion_sort(array):
    for i in range(1, len(array)):
        key = array[i]
        j = i - 1
        while j >= 0 and key < array[j]:
            array[j + 1] = array[j]
            j -= 1
        array[j + 1] = key


if __name__ == '__main__':
    for z in range(50):
        set_array = set2()
        start = time.time() * 1000
        insertion_sort(set_array)
        end = time.time() * 1000
        elapsed = end - start
        print(elapsed)
